### PR TITLE
Lanes: Lane Compare: UI

### PR DESCRIPTION
**Pull request by luvkapur**
- Compare `base` and `target` lanes
 - Show `Component Compare` drawers for each component being `Added` or `Modified`
 - Indicate whether a compare tab has changes
 - Re design Compare Navigation: Add Widgets (Aspects and Changelog)
 - eject deprecated `teambit.component/ui/compare`
 - Responsive Menu Nav: Fix rendering widgets and tabs

### CODE REVIEW

File: .bitmap [hunk 1]
```diff
@@ -435,6 +433,12 @@
         "mainFile": "index.ts",
         "rootDir": "components/entities/doc"
     },
+    "entities/lane-diff": {
```
Comment (luvkapur) [inline .bitmap:436]: review why importing this throws an error on CI

**Review by davidfirst — APPROVED** [review]

File: scopes/lanes/entities/lane-diff/lane-component-diff.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import { ComponentID } from '@teambit/component-id';
```
[the remark this answers is not in the document]
Comment (odedre) [reply scopes/lanes/entities/lane-diff/lane-component-diff.ts:1]: or very mildly

File: scopes/lanes/lanes/lanes.ui.runtime.tsx [hunk 3]
```diff
@@ -75,7 +76,7 @@ export class LanesUI {
             <Route path={LanesModel.lanePath}>
               <Route index element={this.getLaneOverview()} />
               <Route path="~component/*" element={this.getLaneComponent()} />
-              {/* <Route path="~compare/*" element={this.getLanesComparePage()} /> */}
+              <Route path="~compare/*" element={this.getLanesComparePage()} />
```
Comment (odedre) [inline scopes/lanes/lanes/lanes.ui.runtime.tsx]: comment out

File: scopes/lanes/ui/compare/lane-compare-drawer/lane-compare-drawer-name.module.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
+.drawerNameContainer {
+  display: flex;
+  align-items: baseline;
+  padding: 8px 0px;
+  font-size: var(--bit-p-xxs);
```
Comment (odedre) [inline scopes/lanes/ui/compare/lane-compare-drawer/lane-compare-drawer-name.module.scss:5]: fallback color

File: scopes/lanes/ui/compare/lane-compare-loader/lane-compare-loader.module.scss [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.loader {
+  height: 100vh;
```
Comment (odedre) [inline scopes/lanes/ui/compare/lane-compare-loader/lane-compare-loader.module.scss]: problematic and can cause scroll if has header above it or something. 100%?

File: scopes/lanes/ui/compare/lane-compare/lane-compare.tsx [hunk 6]
```diff
@@ -11,55 +10,77 @@ import { ComponentCompareHooks } from '@teambit/component.ui.component-compare.m
...
 export type LaneCompareProps = {
   base: LaneModel;
   compare: LaneModel;
   host: string;
   tabs?: MaybeLazyLoaded<TabItem[]>;
   customUseComponent?: UseComponentType;
+  customUseLaneDiff?: UseLaneDiffStatus;
+  Drawer?: React.ComponentType<LaneCompareDrawerProps>;
+  LaneCompareLoader?: React.ComponentType;
+  ComponentCompareLoader?: React.ComponentType;
```
Comment (odedre) [inline scopes/lanes/ui/compare/lane-compare/lane-compare.tsx:63]: can we add some comments on what each of them does? customUseLaneDiff, Drawer etc

**Review by odedre — APPROVED** [review]

**Review by odedre — APPROVED** [review]

**Pull request merged**